packages/wrangler/src/deploy/deploy.ts

```diff
@@ -220,6 +213,26 @@ export const validateRoutes = (routes: Route[], hasAssets: boolean) => {
 					.join(`\n\n`)
 		);
 	}
+
+	if (mountedAssetRoutes.length > 0 && assets?.directory !== undefined) {
```

Comment:
tiny nit: you've already checked that `assets.directory` is defined.
```suggestion
	if (mountedAssetRoutes.length > 0) {
```

Reply:
I do but ts wasn't happy with it 😅

Reply:
Oh fair enough!